### DISCUSSION TITLE
Share your personal experience with usability of Open Source Scientific Software!

### DISCUSSION THREAD

**Discussion by meagdoh** [meagdoh/ssi-fellowship#12] [General]
This discussion board aims to build off the recent [Open Science Retreat](https://open-science-retreat.gitlab.io/), where scientists, engineers, and open science enthusiasts share their experiences, good and bad, of working with software. 

Share your story! Reply to this thread and tell us:
- What is your role and current context?
- What is the core challenge or opportunity regarding usability in your daily work?
- What are the critical barriers you need to overcome? (or maybe you don’t know yet!)
- Who can help you and the community over this critical barrier?

**Comment by eduardacenteno** [comment]
### When life gives you lemons, make lemonade!



As a PhD student and research assistant in biomedical research, I develop open-source Python data analysis pipelines from scratch and revise colleagues' Python code on GitHub. Both roles require programming discipline, version control, and dependency tracking to ensure project reproducibility and maintenance.

However, **achieving smooth package or code usability can be tricky, especially when collaborating with less experienced colleagues**. Installing packages is a necessary but often tedious process that can take up hours of valuable time. I've found myself stuck in an installation loop, where I spend hours trying to install packages and dependencies before I (or my collaborator) can even start working on the code. **This issue gets even more challenging when packages are not maintained**, limiting their usability and impeding progress. Time is always a limited resource in academic work, so this quest hampers productivity and affects project completion.

This issue is linked to **the low priority and value placed on usability and reproducibility in biological research**. **Biomedical researchers often lack adequate programming training and prioritize publishing scientific papers over code usability**. If usability and reproducibility were prioritized, researchers would be more motivated to invest more time in testing and ensuring interoperability and reusable code. I hope the Open Science movement can help change research software's recognition and reward system, motivating researchers to make their code easy to install and use.

Despite the challenges, I'd like to share a positive experience that resulted in an Open Science prize. During my master's thesis in the Netherlands, I developed [a hands-on tutorial on network and topological data analysis in neuroimaging data](https://github.com/multinetlab-amsterdam/network_TDA_tutorial). After investing significant effort to ensure usability and sustainability, the tutorial is now stable, and people worldwide are using it with no major issues. This work gained significant recognition, leading to **an invitation to publish in a [formal scientific journal](https://link.springer.com/article/10.1007/s00429-021-02435-0) and [a book chapter](https://link.springer.com/chapter/10.1007/978-3-030-80209-7_71)**. Recently, it was even selected for **a prize at the [OSCAwards ceremony](https://spui25.nl/programma/science-for-all-the-2023-oscawards) under the Open Educational Resources category**.

This journey has shown me that **the time invested in ensuring usability is worth it and that a smooth installation process is possible**. More importantly, **this type of work can be recognized and rewarded by academic peers**, leading to excellent outcomes in one's career.
